packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -1055,45 +873,156 @@ function* purchaseWithAnything({
       throw new Error('No solana provider / wallet found')
     }
     const sourceWallet = yield* call(provider.connect)
+    const paymentRouterTokenAccount = yield* call(
+      [
+        sdk.services.paymentRouterClient,
+        sdk.services.paymentRouterClient.getOrCreateProgramTokenAccount
+      ],
+      {
+        mint: 'USDC'
+      }
+    )
+    const externalTokenAccountPublicKey = getAssociatedTokenAddressSync(
+      new PublicKey(inputMint),
+      sourceWallet.publicKey
+    )
 
-    // Swap input mint to usdc
-    console.info('Swapping to USDC then sending to USDC userbank...')
-    yield* call(swapToUsdcAndSendToUserbank, {
-      connection,
-      inputMint,
-      amount: totalAmountWithDecimals,
-      sourceWalletPublicKey: sourceWallet.publicKey,
-      usdcUserBankTokenAccountPublicKey: usdcUserBankTokenAccount.address,
-      signAndSendTransaction: provider.signAndSendTransaction
-    })
+    let transaction: VersionedTransaction
+    if (inputMint === TOKEN_LISTING_MAP.USDC.address) {
+      const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+      const recentBlockhash = yield* call(
+        getRecentBlockhash,
+        audiusBackendInstance
+      )
+      const message = new TransactionMessage({
+        recentBlockhash,
+        payerKey: sourceWallet.publicKey,
+        instructions: [
+          createTransferInstruction(
+            externalTokenAccountPublicKey,
+            paymentRouterTokenAccount.address,
+            sourceWallet.publicKey,
+            totalAmountWithDecimals
+          )
+        ].filter(Boolean) as TransactionInstruction[]
+      })
+
+      transaction = new VersionedTransaction(message.compileToV0Message())
```

Comment:
Would be easier to use `buildTransaction` on `solanaClient` w/  `sourceWallet` as fee payer - can also set priority